server/main.lua

```diff
@@ -20,4 +21,61 @@ for channel, jobs in pairs(restrictedChannels) do
         local player = exports.qbx_core:GetPlayer(source)
         return jobs[player.PlayerData.job.name] and player.PlayerData.job.onduty
     end)
-end
\ No newline at end of file
+end
+
+---@param source number
+local function deleteProp(source)
+    local prop = playerRadios[source]
+
+    if prop then
+        if DoesEntityExist(prop) then
+            DeleteEntity(prop)
+        end
+
+        playerRadios[source] = nil
+    end
+end
+
+---@param source number
+---@return number?
+lib.callback.register('qbx_radio:server:spawnProp', function(source)
+    local ped = GetPlayerPed(source)
+    local coords = GetEntityCoords(ped)
+    local object = CreateObject(`prop_cs_hand_radio`, coords.x, coords.y, coords.z, true, false, false)
+
+    local propExists = lib.waitFor(function()
+        if DoesEntityExist(object) then
+            return true
+        end
+    end, locale('failed_spawn'), 2000)
+
+    if not propExists then return end
```

Comment:
This if statement will always be false, as `waitFor` will error out when timing out, and an error will stop the code from executing further.

Reply:
I'm confused. If it errors out, yes it will be false but if it does infact see the entity exists it returns true?

Reply:
I mean that `not propExists` will never be true and the `return` statement will never be called.

Essentially, I'm calling this line of code useless because it doesn't do anything, since as I said, `waitFor` will error out in the only possible case `not propExists` *could* have been true.

Reply:
That makes more since.

Reply:
Adjusted.